src/vm.rs

```diff
@@ -683,18 +684,52 @@ impl<'a, S: Storage, C: PevmChain> Vm<'a, S, C> {
     }
 
     // Apply rewards (balance increments) to beneficiary accounts, etc.
-    fn apply_rewards(&self, write_set: &mut WriteSet, tx: &TxEnv, gas_used: U256) {
+    fn apply_rewards<DB: Database>(
+        &self,
+        write_set: &mut WriteSet,
+        tx: &TxEnv,
+        gas_used: U256,
+        #[allow(unused_variables)] evm_context: &EvmContext<DB>,
```

Comment:
Or we can put this under the OP feature flag (for now)? Also to be safe in case the Rust compiler doesn't remove the extra unused argument in Ethereum mode.

Reply:
To me it feels like the same mistake when the `optimism` field is added to `struct TxEnv`:

```rust
    // Apply rewards (balance increments) to beneficiary accounts, etc.
    fn apply_rewards<DB: Database>(
        &self,
        write_set: &mut WriteSet,
        tx: &TxEnv,
        gas_used: U256,
        #[cfg(feature = "optimism")] evm_context: &EvmContext<DB>,
    )

----

              self.apply_rewards(
                &mut write_set,
                tx,
                U256::from(result_and_state.result.gas_used()),
                #[cfg(feature = "optimism")]
                &evm.context.evm,
            );
```


Compare the above to the cursed `optimism` field below:

```rust
    fn get_tx_env(&self, tx: Self::Transaction) -> Result<TxEnv, EthereumTransactionParsingError> {
        Ok(TxEnv {
            caller: tx.from,
            gas_limit: tx
                .gas
                .try_into()
                .map_err(|_| EthereumTransactionParsingError::OverflowedGasLimit)?,
            gas_price: get_ethereum_gas_price(&tx)?,
            gas_priority_fee: tx.max_priority_fee_per_gas.map(U256::from),
            transact_to: tx.to.into(),
            value: tx.value,
            data: tx.input,
            nonce: Some(tx.nonce),
            chain_id: tx.chain_id,
            access_list: tx.access_list.unwrap_or_default().into(),
            blob_hashes: tx.blob_versioned_hashes.unwrap_or_default(),
            max_fee_per_blob_gas: tx.max_fee_per_blob_gas.map(U256::from),
            authorization_list: None, // TODO: Support in the upcoming hardfork
            #[cfg(feature = "optimism")]
            optimism: revm::primitives::OptimismFields::default(),
        })
    }
```



-----

Personally, I think a feature flag should only:
1. Add a definition (function, module)
2. Add an arm to an enum

A feature flag should not:
1. Change an interface (including a function signature)

(The feature flags related to OS, CPU can be considered as exceptions. They can conditionally dictate the implementations.)

Reply:
How is adding an arm to an enum different than adding a parameter to a function? You have conditional compilation both at the definition and call sites (matches for `enum`).

If we don't want the extra arms unnecessarily increase all enum variants' memory footprint, then we also don't want the extra function parameters to increase functions' memory footprint when we're sure we don't use them.

Reply:
You just contradicted yourself. If a `match` didn't have a default branch now they need to add one, just like how one need to add `..Default::default()` to the struct constructions. 

The main difference with `TxEnv` is that it is a public (and commonly used) type, so the pollution is annoying. `apply_rewards` is private and only called internally once, which is even less pervasive than the `pub` `RewardPolicy`.

Reply:
> `apply_rewards` is private

ok, this point makes sense